herd/AArch64Sem.ml

```diff
@@ -1564,32 +1573,37 @@ Arguments:
 
       (* Address translation instruction *)
       let do_at op rd ii =
+        check_kvm "AT";
         let open AArch64Base in
         let dir =
           match op.AArch64Base.AT.rw with
           | AArch64Base.AT.W -> Dir.W
           | AArch64Base.AT.R -> Dir.R in
-        let sreg = SysReg (AArch64Base.PAR_EL1) in
+        let sreg = SysReg AArch64Base.PAR_EL1 in
         let mfault ma _ _ =
-          ma >>= M.op1 (Op.ArchOp1 (AArch64Op.SetF)) >>=
-          fun v ->
+            let* a = ma in
+            let* v = M.op1 (Op.ArchOp1 AArch64Op.SetF) a in
             let set_par_el1 = write_reg AArch64Base.par_el1 v ii in
             set_par_el1 >>! B.Next [] in
-        lift_memop rd dir false memtag
-          (fun _ ma _ -> ma >>= (fun a ->
-            M.op1 (Op.ArchOp1 (AArch64Op.SetOA)) a >>= fun v -> write_reg_dest sreg v ii))
-          (to_perms "r" MachSize.Word) (read_reg_ord rd ii) mzero Annot.N ii mfault
+        let mop _ac ma =
+            let* a = ma in
+            let* v = M.op1 (Op.ArchOp1 AArch64Op.SetOA) a in
+            write_reg_dest sreg v ii in
+        let mphy ma a_virt =
+            let ma = get_oa a_virt ma in
+            mop Access.PHY ma >>= M.ignore >>= B.next1T in
+        let ma = read_reg_ord rd ii in
+        let maccess a ma =
```

Comment:
Actually rather than calling do_lift_kvm here, it's much cleaner if we called directly check_ptw. So I went ahead and implemented this change.